pkg/controller/clone/planner.go

```diff
@@ -553,3 +743,65 @@ func createDesiredClaim(namespace string, targetClaim *corev1.PersistentVolumeCl
 
 	return desiredClaim
 }
+
+func createTempSourceClaim(ctx context.Context, namespace string, targetClaim *corev1.PersistentVolumeClaim, snapshot *snapshotv1.VolumeSnapshot, client client.Client) (*corev1.PersistentVolumeClaim, error) {
+	scName, err := getStorageClassNameForTempSourceClaim(ctx, snapshot, *targetClaim.Spec.StorageClassName, client)
+	if err != nil {
+		return nil, err
+	}
+	// Get the appropriate size from the snapshot
+	targetCpy := targetClaim.DeepCopy()
+	restoreSize := snapshot.Status.RestoreSize
```

Comment:
`snapshot.Status` may be nil

Reply:
I can check it here again but in `IsSnapshotValidForClone` we already do some validations to ensure that the required fields are always populated.